x-pack/plugins/fleet/common/authz.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export interface FleetAuthz {
+  fleet: {
+    all: boolean;
+    setup: boolean;
+    readEnrollmentTokens: boolean;
+  };
+
+  integrations: {
+    readPackageInfo: boolean;
+    readInstalledPackages: boolean;
+    installPackages: boolean;
+    upgradePackages: boolean;
+    removePackages: boolean;
+
+    readPackageSettings: boolean;
+    writePackageSettings: boolean;
+
+    readIntegrationPolicies: boolean;
+    writeIntegrationPolicies: boolean;
+  };
+}
+
+interface CalculateParams {
+  fleet: {
+    all: boolean;
+    setup: boolean;
+  };
+
+  integrations: {
+    all: boolean;
+    read: boolean;
+  };
+}
+
+export const calculateAuthz = ({ fleet, integrations }: CalculateParams): FleetAuthz => ({
```

Comment:
This function allows us to have common logic shared across client and server for enforcing access to specific features.